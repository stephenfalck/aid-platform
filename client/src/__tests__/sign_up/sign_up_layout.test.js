import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import SignUpPage from '../../sign_up/sign_up_layout' ;
import SignUpForm from '../../sign_up/sign_up_form';


describe('<SignUpPage>', () => {
  it('renders without crashing', () => {
      shallow(<SignUpPage />);
    });

    it('renders a correct snapshot', () => {
      let wrapper = create(<SignUpPage />)
      expect(wrapper.toJSON()).toMatchSnapshot();
    })

    it('renders sign up form', () => {
      const history = '/history'
      const wrapper = shallow(<SignUpPage history={history}/>);
      const signUpForm = <SignUpForm history="/history"/>;

      //console.log(wrapper.debug())
      expect(wrapper.contains(signUpForm)).toEqual(true);
    });
})