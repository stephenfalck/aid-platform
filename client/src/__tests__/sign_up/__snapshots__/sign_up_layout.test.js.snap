// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPage> renders a correct snapshot 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-align-content-xs-center-13 MuiGrid-justify-xs-center-18 main-container"
  id="sign-up-container"
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-10-39 MuiGrid-grid-sm-8-51"
  >
    <div
      className="MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99"
      id="sign-up-form-paper"
    >
      <h5
        className="MuiTypography-root-125 MuiTypography-h5-141 MuiTypography-gutterBottom-152 MuiTypography-alignCenter-148"
      >
        Please fill out this form and upload a picture of a valid ID
      </h5>
      <form
        id="sign-up-form"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root-161 MuiFormControl-marginNormal-162 MuiFormControl-fullWidth-164"
        >
          <label
            className="MuiFormLabel-root-176 MuiFormLabel-required-181 MuiInputLabel-required-169 MuiInputLabel-root-165 MuiInputLabel-formControl-170 MuiInputLabel-animated-173"
            data-shrink={false}
          >
            First Name
            <span
              className="MuiFormLabel-asterisk-182"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-196 MuiInput-root-183 MuiInput-underline-187 MuiInputBase-fullWidth-205 MuiInput-fullWidth-190 MuiInputBase-formControl-197 MuiInput-formControl-184"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input-206 MuiInput-input-191"
              disabled={false}
              name="first_name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root-161 MuiFormControl-marginNormal-162 MuiFormControl-fullWidth-164"
        >
          <label
            className="MuiFormLabel-root-176 MuiFormLabel-required-181 MuiInputLabel-required-169 MuiInputLabel-root-165 MuiInputLabel-formControl-170 MuiInputLabel-animated-173"
            data-shrink={false}
          >
            Last Name
            <span
              className="MuiFormLabel-asterisk-182"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-196 MuiInput-root-183 MuiInput-underline-187 MuiInputBase-fullWidth-205 MuiInput-fullWidth-190 MuiInputBase-formControl-197 MuiInput-formControl-184"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input-206 MuiInput-input-191"
              disabled={false}
              name="last_name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root-161 MuiFormControl-marginNormal-162 MuiFormControl-fullWidth-164"
        >
          <label
            className="MuiFormLabel-root-176 MuiFormLabel-required-181 MuiInputLabel-required-169 MuiInputLabel-root-165 MuiInputLabel-formControl-170 MuiInputLabel-animated-173"
            data-shrink={false}
          >
            Email
            <span
              className="MuiFormLabel-asterisk-182"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-196 MuiInput-root-183 MuiInput-underline-187 MuiInputBase-fullWidth-205 MuiInput-fullWidth-190 MuiInputBase-formControl-197 MuiInput-formControl-184"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input-206 MuiInput-input-191"
              disabled={false}
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root-161"
          style={
            Object {
              "marginBottom": "8px",
              "width": "100%",
            }
          }
        >
          <label
            className="MuiFormLabel-root-176 MuiFormLabel-required-181 MuiInputLabel-required-169 MuiInputLabel-root-165 MuiInputLabel-formControl-170 MuiInputLabel-animated-173"
            data-shrink={false}
            htmlFor="adornment-password"
          >
            Password
            <span
              className="MuiFormLabel-asterisk-182"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-196 MuiInput-root-183 MuiInput-underline-187 MuiInputBase-formControl-197 MuiInput-formControl-184 MuiInputBase-adornedEnd-201"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input-206 MuiInput-input-191 MuiInputBase-inputType-209 MuiInput-inputType-194 MuiInputBase-inputAdornedEnd-212"
              disabled={false}
              id="adornment-password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
              value=""
            />
            <div
              className="MuiInputAdornment-root-213 MuiInputAdornment-positionEnd-216"
            >
              <button
                aria-label="Toggle password visibility"
                className="MuiButtonBase-root-224 MuiIconButton-root-218"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-223"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-227"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-280"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          className="MuiFormControl-root-161"
          style={
            Object {
              "marginBottom": "8px",
              "width": "100%",
            }
          }
        >
          <label
            className="MuiFormLabel-root-176 MuiFormLabel-required-181 MuiInputLabel-required-169 MuiInputLabel-root-165 MuiInputLabel-formControl-170 MuiInputLabel-animated-173"
            data-shrink={false}
            htmlFor="adornment-password-confirmation"
          >
            Password Confirmation
            <span
              className="MuiFormLabel-asterisk-182"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-196 MuiInput-root-183 MuiInput-underline-187 MuiInputBase-formControl-197 MuiInput-formControl-184 MuiInputBase-adornedEnd-201"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input-206 MuiInput-input-191 MuiInputBase-inputType-209 MuiInput-inputType-194 MuiInputBase-inputAdornedEnd-212"
              disabled={false}
              id="adornment-password-confirmation"
              name="password_confirmation"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
              value=""
            />
            <div
              className="MuiInputAdornment-root-213 MuiInputAdornment-positionEnd-216"
            >
              <button
                aria-label="Toggle password visibility"
                className="MuiButtonBase-root-224 MuiIconButton-root-218"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-223"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-227"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-280"
                />
              </button>
            </div>
          </div>
        </div>
        <label
          aria-describedby={null}
          className=""
          htmlFor="upload"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          title=".jpg / .png / .pdf"
        >
          <span
            className="MuiButtonBase-root-224 MuiButton-root-244 MuiButton-text-246 MuiButton-textPrimary-247 MuiButton-flat-249 MuiButton-flatPrimary-250"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-245"
            >
              Upload ID
            </span>
            <span
              className="MuiTouchRipple-root-280"
            />
          </span>
        </label>
        <input
          accept="image/png, image/jpeg, .pdf"
          id="upload"
          name="image"
          required={true}
          style={
            Object {
              "width": "100%",
            }
          }
          type="file"
        />
        <button
          aria-label="Sign up"
          className="MuiButtonBase-root-224 MuiFab-root-270 MuiFab-extended-274 MuiFab-secondary-273"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "marginBottom": "10px",
              "marginTop": "15px",
              "width": "100%",
            }
          }
          tabIndex="0"
          type="submit"
        >
          <span
            className="MuiFab-label-271"
          >
            Sign up
          </span>
          <span
            className="MuiTouchRipple-root-280"
          />
        </button>
      </form>
    </div>
  </div>
</div>
`;
